apps/compare_requests/views.py

```diff
@@ -99,23 +96,39 @@ def create(self, request, *args, **kwargs):
         else:
             logger.debug('  not found, creating new one...')
 
+            app_merged_branch_name = 'sparrow-test-{}-{}'.format(app_branch_info['sha'], app_dev_info['sha'])
+            logger.debug('Creating merged branch: {}'.format(app_merged_branch_name))
+            try:
+                app_merged_info = github.create_merged_ref(repo_name, app_merged_branch_name, app_branch_info['sha'],
+                                                              APP_TARGET_BRANCH)
+            except:
+                # assume the branch has been correctly created by another process
```

Comment:
You assume but you do not check it in the exception - is that really the only case which can lead to generation of exception? What about IO errors?